cleanlab/multiannotator.py

```diff
@@ -0,0 +1,467 @@
+) -> tuple:
+    """Returns a tuple containing the consensus labels, annotator agreement scores, and quality of consensus
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+
+    pred_probs : np.array
+        An array of shape ``(N, K)`` of model-predicted probabilities, ``P(label=k|x)``.
+        For details, predicted probabilities in the same format expected by the :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+
+    num_annotations : np.array
+        An array of shape ``(N,)`` with the number of annotators that have labeled each example.
+
+    consensus_method : str (Options: ["majority"])
+        The method used to aggregate labels from multiple annotators into a single consensus label. Default is by using the majority vote.
+        For valid consensus methods, view :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`
+
+    quality_method : str
+        Specifies the method used to calculate the quality of the consensus label.
+        For valid quality methods, view :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`
+
+    kwargs : dict
+        Keyword arguments to pass into ``get_label_quality_scores()``.
+
+    Returns
+    ------
+    tuple
+        A tuple of (consensus_label, annotator_agreement, quality_of_consensus).
+    """
+    # Compute consensus label using method specified
+    consensus_label = _get_consensus_label(
+        labels_multiannotator=labels_multiannotator,
+        pred_probs=pred_probs,
+        consensus_method=consensus_method,
+    )
+
+    # Compute the fraction of annotator agreeing with the consensus labels
+    annotator_agreement = _get_annotator_agreement(
+        labels_multiannotator=labels_multiannotator,
+        consensus_label=consensus_label,
+    )
+
+    # Compute the label quality scores of the consensus labels
+    quality_of_consensus = _get_quality_of_consensus(
+        consensus_label=consensus_label,
+        pred_probs=pred_probs,
+        num_annotations=num_annotations,
+        annotator_agreement=annotator_agreement,
+        quality_method=quality_method,
+        **kwargs,
+    )
+
+    return (consensus_label, annotator_agreement, quality_of_consensus)
+
+
+# TODO: update this method once multi-annotator functionality is finalized
+def get_multiannotator_stats(
+    labels_multiannotator: pd.DataFrame,
+    pred_probs: np.array,
+    # label_quality_scores_multiannotator=None, # is this still needed? it is never called in the function
+) -> pd.DataFrame:
```

Comment:
Let's also support `quality_method` as an arg here (with str values). 
For now let's use the current logic `overall_label_health_score` if `quality_method="auto"`.

Let's also support `quality_method="agreement"` in which case we report fraction of this annotator's annotations that agreed with the consensus label.

Reply:
More generally, I think we should support any `quality_method` that can be passed into `get_label_quality_multiannotator()`.

In the case where `quality_method` here is one of the options supported for `get_label_quality_multiannotator()`, we can do the following:

Compute the quality-score of each of one annotator's labels as if it had replaced consensus-label in the consensus-quality-scoring inside of `get_label_quality_multiannotator()`.  Then report the average of the resulting quality-scores over all labels that this annotator provided, as the quality of this annotator.

For example if user specified to use self-confidence label-quality score, you can simply compute the self-confidence of each label provided by one annotator and then report their average to score that annotator.

Reply:
Added `agreement_with_consensus` as a column

TODO: better way to support `quality_method` options, currently there is a mismatch between quality methods for consensus stats and multiannotator stats (eg. `quality_method == 'agreement` makes sense for consensus stats but not multiannotator stats). We might have to separate these two args if the way to aggregate consensus vs annotator quality differs too much.